### ISSUE TITLE
wierd physics problem

### ISSUE THREAD

**Issue by marmust** [marmust/internet-scanner#23]
physics rules:
1: repel nodes in "range" < works well
2: try pulling/pushing connected nodes to "good distance" < doesnt work

(2) explanation:
THIS node - the node from which perspective im writing this
CONNECTED node - the node we are currently pushing/pulling from the perspective of THIS node

(2) should work by first generating a "pull vector3" basically subtract THIS vector3 from CONNECTED vector3
if we just rb.addforce(pull vector3) to CONNECTED node it will fly towards THIS node
if we multiply pull vector3 by a negative value it will fly away.

i will call that value ForceValue (for explanation's sake)

ForceValue will be deicded by a function that takes in the distance between THIS and CONNECTED nodes, and returns:
tanh(distance - DesiredDIstance)

basically if the distance is bigger than DesiredDistance the value will be positive, which will keep the vector on "pull" mode
if the distance is smaller than DesiredDistance then the value will be negative, after multiplying it by the pull vector3, it will become push vector3

alright so whats the problem?
nodes fly towards the -x -y -z quadrant of the world.
i dont know why its happening, but what i do know is that:
if i multiply all that by -1, then the nodes fly towards +x +y +z quadrant.

see NodePhysicsHandler.cs for more info

have a nice day :) thanks

**Comment by Inertia-Squared** [comment]
which mode is this in? I didn't seem to notice this issue in tanh mode, but in regular mode I did notice something like this

**Comment by marmust** [comment]
no not like on ur version, i continued working from your version (you updated while i was working on the physics) and now i have made this bug (it was there even from before the video i just didnt pay attention to it).
you dont have to fix it just if anybody has clues to why this might be happening that would be helpful :)

**Comment by Inertia-Squared** [comment]
Ah fair enough. I'm working on the problem atm, could we create a discord for this repo? I'd love to chat and bounce some ideas as I'm going, I have a few reasons in my head atm for why its not working, but its much easier to show in a vc/screenshare than to tell haha

**Comment by Inertia-Squared** [comment]
Think I fixed it! Running some checks and then I'll make a PR

**Comment by marmust** [comment]
i also worked on it for a bit, also think i fixed it, ill merge both branches :D

**Comment by marmust** [comment]
tysm btw for working on it

**Comment by Inertia-Squared** [comment]
> i also worked on it for a bit, also think i fixed it, ill merge both branches :D

Sounds good! I'm interested to see how our approaches differ haha. Mine is pretty out there, but I've tried to make it as fast as possible for performance.

Always happy to help ^-^

**Comment by marmust** [comment]
i looked at it and yes we did the exact same calculations XD
merged ur version since it had a lookup table (far better perormance)

this problem is pretty much solved so ill close the thread :) again thanks for your help
